### PR TITLE
Create arithmetico_geometric_sequence.py

### PR DESCRIPTION
**Pull request by surdebmalya**
By using it one can calculate everything on Arithmetico-geometric Sequence.

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
```
python3 -m pip install black
black maths/series/arithmetico_geometric_sequence.py
```
Jus like it says in CONTRIBUTING.md.

File: maths/series/arithmetico_geometric_sequence.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    def full_series(self):
+        series = []
+        for i in range(self.n):
+            if i == 0:
+                series.append(str(self.a) + " x " + str(self.b))
+            else:
+                ap_part = str(self.a) + " + " + str(i * self.d)
+                gp_part = str(self.b) + "x" + str(self.r) + "^" + str(i)
+                series.append("(" + ap_part + ")" + " x " + gp_part)
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:53]:
```suggestion
                series.append(f"({ap_part}) x {gp_part})
```

Comment (cclauss) [reply maths/series/arithmetico_geometric_sequence.py:53]: f-strings as discussed in CONTRIBUTING.md.

File: maths/series/arithmetico_geometric_sequence.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    def nth_term_value(self, k):
+        return (self.a + (k - 1) * self.d) * (self.b * pow(self.r, k - 1))
+
+
+def main():
+    run = True
+    while run:
+        try:
+            a = float(input("\nEnter Initial Value For A.P. : "))
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:80]:
```suggestion
            a = float(input("\nEnter Initial Value For A.P. : ").strip())
```
Gracefully deal with leading and/or trailing whitespace in user input as discussed in CONTRIBUTING.md.

File: maths/series/arithmetico_geometric_sequence.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                print("Please Give An Integer For Corresponding Input!")
+        except ValueError:
+            print("Please Give An Integer As The Number Of Terms!")
+
+    ags = AGS(a, d, b, r, n)
+
+    print("\nFull Series : \n{}".format(ags.full_series()))
+    print("\nValue Of Last Term : {:.2f}".format(ags.last_term_value()))
+    print("Sum Of Your A.G.S. : {:.2f}".format(ags.sum()))
+    print("Infinite Series Sum : {:.2f}".format(ags.inf_sum()))
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:123]: f-strings please.

File: maths/series/arithmetico_geometric_sequence.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    not_get_k = True
+    while not_get_k:
+        try:
+            k = float(input("\nValue Of Which Term You Want : "))
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:128]: If you want an integer, why convert it to a float??  Just convert directly to int.

File: maths/series/arithmetico_geometric_sequence.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+python -m doctest -v arithmetic_geometric_sequence.py
+or
+python3 -m doctest -v arithmetic_geometric_sequence.py
+For manual testing run:
+python3 arithmetic_geometric_sequence.py
+"""
+
+
+class AGS:
+    def __init__(self, a, d, b, r, n):
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:15]: Needs type hints.

Single-letter variables are old school.  What about self-documenting parameter names like arithmetic_value, arithmetic_difference, geometric_value, geometric_difference, term_count, k_in_kth_term?  If you can come up with even better names that would be great.

File: maths/series/arithmetico_geometric_sequence.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    ['2.0 x 4.0', '(2.0 + 3.0) x 4.0x0.5^1', '(2.0 + 6.0) x 4.0x0.5^2',
+    '(2.0 + 9.0) x 4.0x0.5^3', '(2.0 + 12.0) x 4.0x0.5^4']
+    Value Of Last Term : 3.50
+    Sum Of Your A.G.S. : 35.00
+    Infinite Series Sum : 40.00
+    Value Of Which Term You Want : 2
+    Value Of 2-th Term : 10.00
+    """
+
+    def full_series(self):
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:45]: Need doctests.

File: maths/series/arithmetico_geometric_sequence.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        for i in range(self.n):
+            if i == 0:
+                series.append(str(self.a) + " x " + str(self.b))
+            else:
+                ap_part = str(self.a) + " + " + str(i * self.d)
+                gp_part = str(self.b) + "x" + str(self.r) + "^" + str(i)
+                series.append("(" + ap_part + ")" + " x " + gp_part)
+        return series
+
+    def last_term_value(self):
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:56]: Need doctests.

File: maths/series/arithmetico_geometric_sequence.py [hunk 8]
```diff
@@ -12,134 +12,198 @@
...
-            if k == int(k):
-                if k <= n:
-                    k = int(k)
-                    not_get_k = False
-                else:
-                    print(
-                        "Please Give Integer Less Than Or Equal To The Numbet Of Terms"
-                    )
+            k_in_kth_term = int(input("\nValue Of Which Term You Want : ").strip())
+            if (k_in_kth_term <= term_count) and (k_in_kth_term >= 0):
```
Comment (ruppysuppy) [inline maths/series/arithmetico_geometric_sequence.py:199]:
```suggestion
            if term_count >= k_in_kth_term >= 0:
```

File: maths/series/arithmetico_geometric_sequence.py (hunk 1 above)
Comment (surdebmalya) [reply maths/series/arithmetico_geometric_sequence.py]: You haven't closed the f-string

File: maths/series/arithmetico_geometric_sequence.py [hunk 9]
```diff
@@ -12,134 +12,198 @@
...
+        >>> AGS(9, 3, 2, -0.5, 0).full_series()
+        []
+        """
         series = []
-        for i in range(self.n):
+        for i in range(self.term_count):
             if i == 0:
-                series.append(str(self.a) + " x " + str(self.b))
+                series.append(
+                    str(self.arithmetic_value) + " x " + str(self.geometric_value)
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:50]:
```suggestion
                    f"{self.arithmetic_value} x {self.geometric_value}"
```

File: maths/series/arithmetico_geometric_sequence.py [hunk 10]
```diff
@@ -12,134 +12,198 @@
...
+                )
             else:
-                ap_part = str(self.a) + " + " + str(i * self.d)
-                gp_part = str(self.b) + "x" + str(self.r) + "^" + str(i)
-                series.append("(" + ap_part + ")" + " x " + gp_part)
+                ap_part = (
+                    str(self.arithmetic_value)
+                    + " + "
+                    + str(i * self.arithmetic_difference)
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:56]:
```suggestion
                    f"{self.arithmetic_value} + {i * self.arithmetic_difference}"
```

File: maths/series/arithmetico_geometric_sequence.py [hunk 11]
```diff
@@ -12,134 +12,198 @@
...
+                )
+                gp_part = (
+                    str(self.geometric_value)
+                    + "x"
+                    + str(self.common_ratio)
+                    + "^"
+                    + str(i)
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:63]:
```suggestion
                    f"{self.geometric_value}x{self.common_ratio}^{i}"
```

File: maths/series/arithmetico_geometric_sequence.py [hunk 12]
```diff
@@ -12,134 +12,198 @@
...
+        """
+        >>> AGS(2, 3, 5, 6, 3).last_term_value()
+        1440
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).last_term_value()
+        1.4976000000000003
+        >>> AGS(9, 3, 2, -0.5, 0).last_term_value()
+        'None'
+        """
+        if self.term_count == 0:
+            return "None"
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:78]: The type hint promises that this function returns a float, not a str.  If we can not calculate a float to return then we could raise a ValueError.

File: maths/series/arithmetico_geometric_sequence.py [hunk 13]
```diff
@@ -12,134 +12,198 @@
...
+        """
+        >>> AGS(2, 3, 5, 6, 3).sum()
+        1600.0
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).sum()
+        8.049600000000002
+        >>> AGS(9, 3, 2, -0.5, 0).sum()
+        'None'
+        """
+        if self.term_count == 0:
+            return "None"
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:93]: The type hint promises that this function returns a float, not a str.  If we cannot calculate a float to return then we could raise a ValueError.

File: maths/series/arithmetico_geometric_sequence.py [hunk 14]
```diff
@@ -12,134 +12,198 @@
...
+        """
+        >>> AGS(2, 3, 5, 6, 3).kth_term_value(2)
+        150
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).kth_term_value(1)
+        3.9000000000000004
+        >>> AGS(9, 3, 2, -0.5, 0).kth_term_value(0)
+        'None'
+        """
+        if k_in_kth_term == 0:
+            return "None"
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:136]: The type hint promises that this function returns a float, not a str.  If we cannot calculate a float to return then we could raise a ValueError.

File: maths/series/arithmetico_geometric_sequence.py [hunk 15]
```diff
@@ -12,134 +12,198 @@
 
 
 class AGS:
```
Comment (cclauss) [inline maths/series/arithmetico_geometric_sequence.py:14]:
```suggestion
class ArithmeticGeometricSequence:
```

**Pull request closed without merging**